cmake/Modules/FindMKLDNN.cmake

```diff
@@ -18,6 +18,48 @@ IF(NOT MKLDNN_FOUND)
 
   SET(IDEEP_ROOT "${PROJECT_SOURCE_DIR}/third_party/ideep")
   SET(MKLDNN_ROOT "${PROJECT_SOURCE_DIR}/third_party/ideep/mkl-dnn")
+
+  if(USE_XPU) # Build oneDNN GPU library
+    if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+      set(DNNL_HOST_COMPILER "g++")
+      # g++ is soft linked to /usr/bin/cxx, oneDNN would not treat it as an absolute path
+    else()
+      set(DNNL_HOST_COMPILER DEFAULT)
```

Comment:
Please remove the default path as its behavior is undefined.

Reply:
Has modified this part, XPU oneDNN would not build with compiler other than gnu g++, we may remove the restriction after oneDNN community supports others.